### PR TITLE
new group field for info about adding funds

### PR DESCRIPTION
**Pull request by jlopker**
@chime-mu Can you push this to staging?

### CODE REVIEW

**Comment by chime-mu** [comment]
@jlopker It's not really clear what you want me to do here since you're asking me to push two different branches to staging at the same time?

**Comment by chime-mu** [comment]
@jlopker I've deployed this PR to stage. I'm thinking we can do PR-159 after this.

**Comment by jlopker** [comment]
Ok great! Yea sorry for not realizing that 😬 I’ll let you know when this
is ready for prod and we can push the branding update to stage.

File: api/db/migrate/20180328223945_add_add_funds_to_groups.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddAddFundsToGroups < ActiveRecord::Migration
+  def change
+    add_column :groups, :add_funds, :string
```
Comment (mixmix) [inline api/db/migrate/20180328223945_add_add_funds_to_groups.rb]: should this be `:string` or `:text` ?   I'm a bit rusty but memory and a quick search suggests `:string` has a length constraint (perhaps 255).

I'll see if I can break it on staging !

File: api/scripts/migrate [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#
+sudo docker run -e DATABASE_URL=<DB_URL> greaterthanfinance/cobudget-api-stage rake db:migrate
```
Comment (mixmix) [inline api/scripts/migrate]: This looks like it needs attention. Other files in this folder look like : 

```
#! /bin/bash
cd /cobudget-api; env -i $(cat /root/env-vars) rake cobudget:deliver_recent_activity_emails
```

Things I notice that are different here:
- there's a missing `#! /bin/bash`
- this script seems to be locked to a staging api (cobudget-api-stage)?
  - I've not read about what docker is doing in this project, so could be off

File: ui/app/components/admin-page/admin-page.html [hunk 3]
```diff
@@ -33,6 +33,11 @@ <h4 class="admin-page__group-list-subheader">Edit {{ group.name }}</h4>
           <div class="admin-page__input-help-text">i.e. In this group, members of {{ group.name }} fund projects to improve their work environment and support each other.</div>
           <a class="admin-page__description-markdown-link" href="https://www.loomio.org/markdown" target="_blank">formatting help</a>
         </md-input-container>
+        <md-input-container class="admin-page__input md-block">
+          <label>Add Funds</label>
+          <input ng-model="group.addFunds" type="url"/>
+          <div class="admin-page__input-help-text">A url of a page with info about adding funds to an account (include http(s)://).</div>
```
Comment (mixmix) [inline ui/app/components/admin-page/admin-page.html]: is this going to be a url or MD block which can have a link in it?
Will try out the staging server and see how this behaves.

I think the `include http(s)` is probably going to confuse people. If this is only ever going to be a link then maybe we shouldn't use MD, and instead should just turn this into an `<a>` and validate the input is a link with `http(s)://www` ..

Comment (mixmix) [reply ui/app/components/admin-page/admin-page.html]: I think the https thing is going to confuse some people, might need so validation when they get it wrong / right?

File: ui/config/stage.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  apiPrefix: "https://stage.cobudget.co/api/v1"
+}
```
Comment (mixmix) [inline ui/config/stage.js]: there's a `ui/config/staging.js` which has this in it: 

```js
module.exports = {
  apiPrefix: "https://api.cobudget.co/api/v1"
}
```

Feels like this is some duplication which needs resolving?

**Comment by mixmix** [comment]
# Smoke test/ UI feedback

:+1: Works as advertised!

Here are some things which I think minor tweaks would make clearer: 

## Settings page 

### 1. Help non technical users with "URL"

selection_561

Not everyone everyone know what a URL is. I think the stuff about http should be in the rror message, not in the helper text : 
 ```
Must be a url (this means is needs to include http:// or https:// at the start)
```

### 2. Minor style things

selection_559

The toggle thing seems inverted to me. The words say "off" but the switch looks deactivated. This could be a seperate issue/ ticket

## Group page

selection_563

The semantics of all this data are getting a bit cluttered for me with 3 things. To me it reads: "your funds are 348, also the group has 4,814 and oh also if you want to add funds you can do that". I'm a bit confused if I'm adding funds to the group or to my cache or what. Maybe that's fine.
I think the things should be a seperated a bit (this could also be just tests on live humans and see what they say. Mix is not a designer!)

## Manage Funds page

selection_562

As a naive user who has missed the subtle link to `Add funds` I might go to the Manage funds Page. Hmmm this is only accessible to Admins, but to me seems like we could have an "In" and "Out" sections on that page now. This could be some other future maybe ticket

File: ui/config/stage.js (hunk 4 above)
Comment (chime-mu) [reply ui/config/stage.js]: @mixmix True. The only 3 configs that are actually in use is `default.js` (or `development.js`), `stage.js` and `production.js`. The reason for the duplication is a recent move in server setup. Feel free to delete `staging.js` - and and let me know if you need more details.

File: api/scripts/migrate (hunk 2 above)
Comment (jlopker) [reply api/scripts/migrate]: @chime-mu Is this part of the server setup changes? Is this line supposed to be removed?

Comment (chime-mu) [reply api/scripts/migrate]: @mixmix @jlopker 
* It's safe to add `#! /bin/bash` to the `migrate` script - but hardly necessary. The script is currently more of a reminder for me (that I wanted to check in) about how to do a migration when everything is running in a docker container. 
* Yes, this script is for update using the stage api docker image, but can be used for a production image as well. Again, it's a work in progress and will be consolidated into a consistent documented deployment pipeline.

Comment (chime-mu) [reply api/scripts/migrate]: The `activity_emals` script is a script to be executed by `cron` so it won't function unless the executable is specified. Hence `#! /bin/bash`

**Comment by jlopker** [comment]
@chime-mu Can you make sure `api/scripts/migrate` is correct? I'm not sure why it's different on this branch than on master. If it's good, can you deploy to production? Thanks

**Pull request merged**